packages/core/src/_common/base/api_middleware.js

```diff
@@ -20,19 +105,37 @@ class APIMiddleware {
         const promise = promiseRejectToResolve(response_promise);
 
         const key = requestToKey(request);
+        const request_type = this.getRequestType(request);
+
+        if (request_type) performance.mark(`${request_type}_start`);
 
         if (options.callback) {
             promise.then(options.callback);
         }
 
         this.debounced_calls[key] = promise;
 
-        promise.then(() => {
+        promise.then(response => {
+            const response_type = this.getRequestType(response);
+            if (response_type) this.defineMeasure(response_type);
+
             delete this.debounced_calls[key];
         });
 
         return promise;
     }
+
+    sendRequestsStatistic = () => {
```

Comment:
this name is present `sendRequestsStatistic` in bot-web-ui so if the user visits bot then there are chances the function can get overridden or error, I am not sure. we need to check this and handle it.